### ISSUE TITLE
GenAI Roadmap

### ISSUE THREAD

**Issue by arun2code** [arun2code/BookYourShow#1]
Roadmap To Learn Generative AI In 2024
Prerequisites
1. Python Programming Language -1 Month
Python:
python-logo-master-v3-TM-flattened

Complete Python Playlist In English: YouTube

Complete Python Playlist In Hindi: YouTube

Flask Playlist: YouTube

Fast API Tutorials YouTube

2. Basic Machine Learning Natural Language Processing (Day 1 - Day 5) YouTube
Why NLP?
One hot Encoding, Bag Of Words,
TFIDF
Word2vec,AvgWord2vec
3. Basic Deep Learning Concepts (Day 1- Day 5) YouTube
ANN - Working Of MultiLayered Neural Network
Forward Propogation, Backward Propogation
Activation Functions, Loss Functions
Optimizers
4. Advanced NLP Concepts (Day 6 - Last Video) YouTube
RNN, LSTM RNN
GRU RNN
Bidirection LSTM RNN
Encoder Decoder, Attention is all you need ,Seq to Seq
Transformers
5. Starting the Journey Towards Generative AI (GPT4,Mistral 7B, LLAMA, Hugging Face Open Source LLM Models,Google Palm Model)
image

OpenAI YouTube YouTube
image

Langchain Tutorials With Projects YouTube YouTube

Chainlit YouTube

Google Gemini YouTube

5. Vector Databases And Vector Stores
ChromaDB
FAISS vector database, which makes use of the Facebook AI Similarity Search (FAISS) library
LanceDB vector database based on the Lance data format
Cassandra DB For storing Vectors
6. Deployment Of LLM Projects
AWS
Azure
LangSmith
LangServe
HuggingFace Spaces
